manage.sh

```diff
@@ -46,14 +49,74 @@ function run-devenv {
          $CONTAINER
 }
 
+function test-devenv {
+    echo "Testing frontend..."
+    test-devenv-frontend || exit 1;
+    echo "Testing backend..."
+    test-devenv-backend || exit 1;
+}
+
+function test-devenv-frontend {
+    # TODO Add frontend unit tests call
+    echo "Not yet implemented!"
+
+#    if ! $(docker images | grep $IMGNAME | grep -q $REV); then
+#        build-devenv
+#    fi
+#
+#    CONTAINER=$IMGNAME:latest
+#
+#    echo "Running development image $CONTAINER to test backend..."
+#    docker run -ti --rm \
+#           -w /home/uxbox/uxbox/backend \
+#           -v `pwd`:/home/uxbox/uxbox  \
+#           -v $HOME/.m2:/home/uxbox/.m2 \
+#           -e UXBOX_API_URL="/api" \
+#           -e UXBOX_VIEW_URL="/view" \
+#           $CONTAINER ./scripts/build-tests.sh
+}
+
+function test-devenv-backend {
+    # TODO Add backend unit tests call
+    echo "Not yet implemented!"
+
+#    if ! $(docker images | grep $IMGNAME | grep -q $REV); then
+#        build-devenv
+#    fi
+#
+#    CONTAINER=$IMGNAME:latest
+#
+#    echo "Running development image $CONTAINER to test frontend..."
+#    docker run -ti --rm \
+#           -w /home/uxbox/uxbox/frontend \
+#           -v `pwd`:/home/uxbox/uxbox  \
+#           -v $HOME/.m2:/home/uxbox/.m2 \
+#           -e UXBOX_API_URL="/api" \
+#           -e UXBOX_VIEW_URL="/view" \
+#           $CONTAINER ./scripts/build-tests.sh
+}
+
 function build-release-frontend-local {
+    if ! $(docker images | grep $IMGNAME | grep -q $REV); then
+        build-devenv
+    fi
+
+    mkdir -p $HOME/.m2
+    rm -rf ./frontend/node_modules
+    # FIXME Ugly... should be a better way
+    chmod 777 ./frontend
```

Comment:
why this `chmod 777` ? do you have permissions related problem with that mounted directories?

Consider building the image with `--build-arg EXTERNAL_UID=$(id -u) ` (i will add it to the default build-devenv command. But this parameter will create the image user with the same id as in your machine, so your permission problems will disappear.

Reply:
Indeed I've had permissions issues when building the frontend (don't remember which files exactly though) so I added the FIXME as a reminder.

Your idea should probably do the trick. I will try it a bit later this week and let you know.